### PR TITLE
Adds master

### PR DESCRIPTION
**Pull request by ezmiller**
@AnnJohn @pletcher this is ready for review

### CODE REVIEW

File: test/b_models_test.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          'author',
+        ]);
+        mockPost.id = post.get('id');
+        done();
+      });
+  });
+
+  it('Comments model exists', (done) => {
+    expect(blog.Comments, 'to be defined');
+    done();
```
Comment (pletcher) [inline test/b_models_test.js]: Minor nit: for synchronous tests, should we omit `done` altogether?

File: test/b_models_test.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      .catch((err) => { done(err); })
+      .then((post) => {
+        expect(post.attributes, 'to have keys', [
+          'title',
+          'body',
+          'id',
+          'author',
+        ]);
+        mockPost.id = post.get('id');
+        done();
```
Comment (pletcher) [inline test/b_models_test.js]: Should this be in the `then` callback starting on line 59?

File: test/b_models_test.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      .save()
+      .catch((err) => { done(err); })
+      .then((post) => {
+        expect(post.attributes, 'to have keys', [
+          'title',
+          'body',
+          'id',
+          'author',
+        ]);
+        mockPost.id = post.get('id');
```
Comment (pletcher) [inline test/b_models_test.js:64]: Why are set `mockPost`'s `id`?

File: test/b_models_test.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      .then((usr) => {
+        expect(usr.attributes, 'to have keys', [
+          'name',
+          'email',
+          'username',
+        ]);
+        expect(usr.get('name'), 'to be', mockUser.name);
+        expect(usr.get('email'), 'to be', mockUser.email);
+        expect(usr.get('username'), 'to be', mockUser.username);
+        mockUser.id = usr.get('id');
```
Comment (pletcher) [inline test/b_models_test.js]: Why set the `id`

File: test/c_blog_server_test.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
+"use strict";
+
+const expect  = require('unexpected');
+const request = require('supertest');
+const baseUrl = 'http://localhost:3000';
+
+// Set mode to testing.
+process.env.TESTING = true;
```
Comment (pletcher) [inline test/c_blog_server_test.js]: It looks like the package.json `test` script already sets `process.env.NODE_ENV = 'testing'` -- does this do something else?

File: test/root_hooks.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+"use strict";
+
+const blog = require('../blog_server');
+
+before((done) => {
+  blog.up().then(done);
+});
+
```
Comment (pletcher) [inline test/root_hooks.js:8]: I'm wondering if we should add some database cleanup/teardown around each test. Unfortunately, Mocha doesn't give us access to a given test's environment (or even the test to run), so something like [this](http://stackoverflow.com/questions/27220269/how-do-i-perform-an-around-or-aroundeach-with-mocha) might be the cleanest approach (and we could ditch that nasty `bind` if we used an arrow function, at least).

**Comment by pletcher** [comment]
This looks pretty good! I added fixed some minor things in the README and added some comments to the tests. Once we figure out those comments, this'll be ready to go!

File: test/b_models_test.js (hunk 1 above)
Comment (ezmiller) [inline test/b_models_test.js]: Done.

**Comment by ezmiller** [comment]
@pletcher I've fixed most if not all of the README problems here. I should now just go through and properly isolate the tests in the way that I did for the social blog lab.

**Comment by pletcher** [comment]
That isolation sounds like the right thing to do to me.

Will you be able just to copy-paste the setup/teardown methods from there?

Once that's in, we can :ship:!

**Comment by ezmiller** [comment]
@pletcher, It wasn't actually cut & paste, but it went quickly. Just had to replace the login fn stuff since there's no user validation here yet. And a few other things. Anyway, I've pushed the new isolated tests here, so should be ready to go.

File: test/c_blog_server_test.js [hunk 7]
```diff
@@ -24,156 +23,262 @@ let mockComment = {
   body: 'This is just a test comment.'
 };
 
+const cleanup = () => {
```
Comment (pletcher) [inline test/c_blog_server_test.js:26]: :heart:

**Comment by pletcher** [comment]
:shipit:

**Comment by ezmiller** [comment]
@pletcher some error above is preventing merge here. But this is ready from my point of view.

**Pull request merged**